pyplis/calib_base.py

```diff
@@ -190,10 +190,11 @@ def calib_coeffs(self, val):
                                  "in current optimisation function. "
                                  "Please check and update class attribute "
                                  "calib_fun first...")
-        if len(self._calib_coeffs) > 0:
-            warn("Setting calibration coefficients manually. This may introduce "
-                 "analysis errors. It is recommended to use the method "
-                 "fit_calib_data instead")
+        if self._calib_coeffs is not None:
+            if len(self._calib_coeffs) > 0:
+                warn("Resetting calibration coefficients manually. This may introduce "
```

Comment:
I think `if self._calib_coeffs and len(self._calib_coeffs) > 0:` would do the same and avoid nested ifs, but I might be wrong

Reply:
The problem is, that `None` has no length. So if `self._calib_coeffs` is None, a combined if-clause would throw a `TypeError: object of type 'NoneType' has no len()`.
However you can replace `if self._calib_coeffs is not None`  with `if self._calib_coeffs`. I personally prefer the explixit form but both works fine.

Reply:
Okay, sorry I was not aware that the second condition would not be evaluated in that case. Sure then your one-line is more concise.